sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -233,6 +234,60 @@ protected void resolveAllCallsFromSourceMethod(
                                 sourceMethod.getSignature(), targetMethod, stmt, cg, workList)));
   }
 
+  Set<MethodSignature> callSources = new HashSet<>();
+
+  /**
+   * It resolves the start-run implicit calls caused by the given source method
+   *
+   * @param sourceMethod the inspected source method
+   * @param cg implicit start-run calls will be added to the call graph
+   * @param workList new run methods will be added to the work list
+   */
+  protected void implicitStartRunCall(
+      SootMethod sourceMethod, MutableCallGraph cg, Deque<MethodSignature> workList) {
+    for (Stmt stmt : sourceMethod.getBody().getStmts()) {
+      if (!stmt.isInvokableStmt()) {
+        continue;
+      }
+      InvokableStmt invokableStmt = stmt.asInvokableStmt();
+      if (invokableStmt.getInvokeExpr().isEmpty()) {
+        continue;
+      }
+      MethodSignature methodSig = invokableStmt.getInvokeExpr().get().getMethodSignature();
+      if (!methodSig.getType().equals(VoidType.getInstance())
+          || !methodSig.getParameterTypes().isEmpty()
+          || !methodSig.getName().equals("start")) {
+        continue;
+      }
+      Optional<MethodSignature> cMethodSigOpt = resolveConcreteDispatch(view, methodSig);
+      if (cMethodSigOpt.isEmpty()) {
+        continue;
+      }
+      MethodSignature cMethodSig = cMethodSigOpt.get();
+      callSources.addAll(cg.callSourcesTo(cMethodSig));
+      if (callSources.isEmpty()) {
+        return;
+      }
```

Comment:
why are you doing this?

Reply:
you should have already created an edge when you created the calls that you get by calling callTo